### PR TITLE
SP2-572 - Added add-steps

### PR DESCRIPTION
**Pull request by Thomas-Geraghty**
- Added add-steps controller, routing, view
- Fixed minor bug in interpolate function which meant arrays would be parsed back into objects
- Updated Step and NewStep type - the API still expects actor[] and actorOptionId? this needs cleaning up

### CODE REVIEW

File: server/routes/add-steps/locale.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "en": {
+    "page": {
+      "title": "Add steps to goal - {{ applicationName }}",
+      "errorTitle": "Error: Add steps to goal - {{ applicationName }}"
+    },
+    "mainHeading": {
+      "title": "Add steps to {{ subject.givenName }} goal",
+      "caption": "{{ goal.title }}"
+    },
+    "detailHeading": "View information from John’s assessment",
```
Comment (alexw-moj) [inline server/routes/add-steps/locale.json]: John's?

**Review by pk2453 — DISMISSED** [review]

File: server/views/pages/add-steps.njk [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+<a href="#" class="govuk-back-link">Back</a>
+<div class="govuk-grid-row">
+    <div class="govuk-grid-column-full">
+        <span class="govuk-caption-l">{{ locale.mainHeading.caption }}</span>
+        <h1 class="govuk-heading-l">{{ locale.mainHeading.title }}</h1>
+
+        {{ govukDetails({
+            summaryText: locale.detailHeading,
+            text: "...."
+        }) }}
```
Comment (pk2453) [inline server/views/pages/add-steps.njk:49]: I think it was a placeholder for content which I forgot to add. Will add this on Monday

File: server/routes/add-steps/locale.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          "Someone else (include who in the step)"
+        ]
+      }
+    },
+    "saveButtonText": "Save and continue",
+    "addAnotherButton": "Add another step",
+    "removeStepLinkButton": "Remove",
+    "errors": {
+      "step-description": {
+        "isNotEmpty": "You need to enter a step"
```
Comment (alexw-moj) [inline server/routes/add-steps/locale.json]: `Select or enter what they should do to achieve the goal` is the text given on the errors spreadsheet. Also, should that be `isEmpty`?

File: server/routes/add-steps/locale.json (hunk 1 above)
Comment (Thomas-Geraghty) [reply server/routes/add-steps/locale.json]: Damnit, knew I missed one!

Comment (Thomas-Geraghty) [reply server/routes/add-steps/locale.json]: Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/110/commits/f9ae1c50f0000ebc0a4777409f1e272c809a7029

File: server/routes/add-steps/locale.json (hunk 3 above)
Comment (Thomas-Geraghty) [reply server/routes/add-steps/locale.json]: Nice spot, resolved the error message in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/110/commits/dee2b6aa46840ca20fc9747033918fe91438bd7e.

And surprisingly not, it's a bit long to explain but basically, 
- a failed validation constraint from our validation middleware comes through like
```
[
  "step-description": {
   "isNotEmpty": true
  }
]
```
where the failed constraint is set to 'true'
- if we get 'isNotEmpty` as a failed constraint, it'll match to this to the same named error in the locale file

fyi - I'm aware this is pretty unclear, I've not actually come up with a good solution on flipping this, even though I've spent quite a bit of time on it when I first developed the solution for a previous project.

File: server/views/pages/add-steps.njk (hunk 2 above)
Comment (pk2453) [reply server/views/pages/add-steps.njk:49]: added summary text

**Pull request closed without merging**